aws-rds-dbinstance/aws-rds-dbinstance.json

```diff
@@ -447,7 +447,9 @@
     "/properties/OptionGroupName": "$lowercase(OptionGroupName)",
     "/properties/SourceDBInstanceAutomatedBackupsArn": "$lowercase(SourceDBInstanceAutomatedBackupsArn)",
     "/properties/SourceDBInstanceIdentifier": "$lowercase(SourceDBInstanceIdentifier)",
-    "/properties/MasterUserSecret/KmsKeyId": "$join([\"arn:(aws)[-]{0,1}[a-z]{0,2}[-]{0,1}[a-z]{0,3}:kms:[a-z]{2}[-]{1}[a-z]{3,10}[-]{0,1}[a-z]{0,10}[-]{1}[1-3]{1}:[0-9]{12}[:]{1}key\\/\", KMSKeyId])"
+    "/properties/MasterUserSecret/KmsKeyId": "$join([\"arn:(aws)[-]{0,1}[a-z]{0,2}[-]{0,1}[a-z]{0,3}:kms:[a-z]{2}[-]{1}[a-z]{3,10}[-]{0,1}[a-z]{0,10}[-]{1}[1-3]{1}:[0-9]{12}[:]{1}key\\/\", KmsKeyId])",
```

Comment:
What will happen if customer entered KmsKey full Arn instead of KmsKey simple alias? Do CFN neglect applying `propertyTransform` function as if fails to match first regex expression in join?

Reply:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-stack-drift.html

at the moment this is considered a platform limitation

Reply:
@moataz-mhmd from the documentation (https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-model-false-drift.html): 

```
The property transform provides a way for CloudFormation to accurately compare the resource property value specified in the template with the value returned from the read handler. During drift detection, if CloudFormation finds a property where the template value differs from the value returned by the read handler, it determines if a property transform has been defined for that property in the resource schema. If it has, CloudFormation applies that property transform to the value specified in the template, and then compares it to the read handler value again. If these two values match, the property isn't considered to have drifted, and is marked as IN_SYNC
```

I.e. it first would compare the actual value (expanded ARN) and it is expected to match immediately. I.e. propertyTransform should not be engaged at all.

Reply:
So If I understood it correctly, We consider these two inputs are equal
i.g:
```
KmsKeyId: mySecret1
MasterUserSecret:
   KmsKeyId: arm:aws:us-east-1:kms:antherSecret
```  
is equal to 

```
KmsKeyId: mySecret1
MasterUserSecret:
   KmsKeyId: arm:aws:us-east-1:kms:mySecret1
```

Reply:
@moataz-mhmd not really. The KmsKey alias drift is a known case.